src/data/enrollments/EnrollmentsD2Repository.ts

```diff
@@ -0,0 +1,69 @@
+import _ from "lodash";
+
+import { D2Api } from "types/d2-api";
+
+import { Id } from "domain/entities/Base";
+import { Async } from "domain/entities/Async";
+import { Enrollment } from "domain/entities/enrollments/Enrollment";
+
+import {
+    EnrollmentsRepository,
+    EnrollmentsRepositoryParams,
+    TrackerResponse,
+} from "domain/repositories/enrollments/EnrollmentsRepository";
+
+export class EnrollmentsD2Repository implements EnrollmentsRepository {
+    constructor(private api: D2Api) {}
+
+    async getList(params: EnrollmentsRepositoryParams, enrollmentsIds: Id[]): Async<Enrollment[]> {
+        const { programId, orgUnitId } = params;
+
+        const idList = enrollmentsIds.join(";");
+
+        // NOTE: looks like @eyeseetea/d2-api/api/trackerEnrollments.d.ts: Params is not interpreted correctly and marked as required
+        const { instances: enrollments } = (await this.api.tracker.enrollments
+            .get({
+                program: programId,
+                orgUnit: orgUnitId,
+                ouMode: "SELECTED",
+                fields: { enrollment: true, status: true },
+                enrolledBefore: `${new Date().toISOString().replace("Z", "")}`,
+                enrollment: idList,
+                skipPaging: true,
+            })
+            .getData()) as { instances: Enrollment[] };
```

Comment:
This will not be necessary when https://github.com/EyeSeeTea/d2-api/pull/151 is merged, as it infers the enrollments type from the fields. Let's keep as is.